### ISSUE TITLE
Speech bubbles hidden behind sprites and wrongly positioned.

### ISSUE THREAD

**Issue by sfederici** [jmoenig/Snap--Build-Your-Own-Blocks#1460]
Scratch's speech bubbles are drawn always in foreground, so that the bubble is always clearly visible (this latter behaviour is correct, as you should want that your bubbles are clearly visible). Snap's speech bubbles instead are drawn taking into account the layer of the sprite. So they can appear behind other sprites when the speaking sprite is behind other sprites. 

There is another difference between Snap's and Scratch's speech bubbles. Scratch's speech bubbles are positioned so that they do not hide the sprite (whenever possible). Snap's speech bubbles instead are drawn by positioning the speech bubble's tail close to the sprite's rotation center, so that they often hide part of the sprite.

[speech bubble test.zip](https://github.com/jmoenig/Snap--Build-Your-Own-Blocks/files/532529/speech.bubble.test.zip)

**Comment by jmoenig** [comment]
Hi Stefano. You're precisely describing Snap's speech bubble behavior and how it differs from Scratch. This was actually what I wanted, but I've since realized that speech bubble positioning and layering sometimes causes conflicts and may need to be thought over. Thanks for highlighting this!

**Comment by sfederici** [comment]
Thanks Jens. Let me know if and when you will change this behaviour. I would like to switch to Snap for my courses but I would like to keep all the slides and tutorials that I already prepared for Scratch. I tested a fair amount of projects created by my students and the only differences I noted until now is the behaviour of the bubbles and the default "arrow" costume that shows up when my students go "backwards" by using "switch to costume (costume #) - 1".

**Comment by brianharvey** [comment]
> and the default "arrow" costume that shows up when my students go "backwards" by using "switch to costume (costume #) - 1".

When I first saw this I thought "that's wrong!" but it would be a problem in Snap<i>!</i> only if it were the behavior of a "previous costume" primitive.  Of course the "switch to costume" block doesn't know that the zero it got as input was an attempt to move backward from costume 1.  You can and should fix this for yourself by writing a "previous costume" custom block that checks for current costume 1 and moves to costume LENGTH OF (WARDROBE) instead of costume 0.

**Comment by sfederici** [comment]
I agree that the problem could be easily corrected by improving your own programming habits. But this creates some trouble when you import a project from Scratch 2 to Snap, for example by using Snapin8r. It would be a cool nice feature of Snap that Scratch projects can be imported straightforwardly without having to change the way they works.
